### DISCUSSION TITLE
Universal Adapter Build: GameCube Adapter Assembly Questions

### DISCUSSION THREAD

**Discussion by alastairbdoggett** [darthcloud/BlueRetro#82] [Q&A]
Starting the process of making a universal adapter and the first target console adapter i'm trying is GameCube. Following the [GC cable schematic](https://github.com/darthcloud/BlueRetroHW/blob/master/DIY/GameCube.pdf) I have two questions:

1. When [building a GameCube only adapter](https://github.com/darthcloud/BlueRetro/discussions/63) (i.e. using `BlueRetro_gc_spiffs.bin`) I used 6 connections directly to the ESP32 board: 1xGND, 1x5V and 4xDATA -- is it possible to use the same 6 connections onto the DB25 male adapter?

2. In the Assembly Instructions it says
```
If using an universal FW, make sure to connect I39 to GND and IO21, IO22, IO25, IO32, IO33, I34 & I35 to GND.
```
Using the GC cable schematic I get:

I39 -> pin 22 -> GND
IO21 -> pin 3 -> GND
IO25 -> pin 12 -> GND
IO32 -> pin 11 -> GND
IO33 -> pin 24 -> GND
I34 -> pin 10 -> GND
I35 -> pin 23 -> GND

Do I need to attach all the above pins to GND? I'm struggling to see how I would do that since I only get at most 4x GND wires from the 4x GC controller cables.

**Comment by darthcloud** [comment]
Yes its important to connect all pins as described for console detection to work on universal build.

GND is GND, connect them all together on DB25 side and then connect what you got on it. You don't need individual connection, they just need a path to it.

**Reply by alastairbdoggett** [reply]
OK this makes sense! Especially since I made a counting error in my previous post 😳 

Indeed, each GC controller cable gives 2x GND so we get the desired 8x GND that are needed (I39, IO21, IO22, IO25, IO32, IO33, I34 & I35)

Having wired all of this up I believe I am almost there! When I plug the GC adapter into the universal adapter and **power on via USB cable** I am able to see logs in BlueRetro Web Console and pair a DS4 controller successfully.

However, when I **power on via GC** (i.e. by plugging the cables into the console), I see the red LED on the ESP32 come on but am unable to find BlueRetro in the Web Console and unable to pair a DS4 controller -- everything else seems OK with the system idling on the menu without a problem.

Edit: Perhaps I may have made a mistake wiring 5V? I only have a single 5V connected, namely, from P1 controller (i.e. pin 7 on DB25) because there is only a single 5V available on the [cable schematic](https://github.com/darthcloud/BlueRetroHW/blob/master/DIY/GameCube.pdf).

Edit: I also have pins 5, 8 & 9 not connected to a GND because I don't have any more GND wires from the GC cables!

Any ideas what might be going wrong here? And do I need to connect **all 5V** to the same pin 7 on my DB25?

**Comment by alastairbdoggett** [comment]
Update: Disconnecting the 5V from P1 (i.e. leaving pin 7 open), plugging the cables into the GC and powering the ESP32 from USB, I can successfully pair a DS4 controller and see the logs in the Web Config. There is unplayable lag and I can't consistently connect more than a single player. I can navigate the GC so it appears to have detected the console correctly?

**Comment by alastairbdoggett** [comment]
Another Update: Summarizing the testing i've been doing in case it helps :)

- I am able to pair 4x DS4 controllers to **only the universal adapter** using USB power;
- Still using USB power, if I connect the GC adapter and insert the cables to the GC, when I power on the console I get very inconsistent pairing + response (sometimes it will pair with no response, sometimes it won't even pair)
- Using 5V from P1 onto DB25 pin 22  gives red LED on ESP32 (assuming power is therefore getting to it) but not able to locate it in the Web Config (thus no pairing either)

Makes me think I have screwed something up on the GC adapter side given, when not using this, I can get logs in the Web Console and connect all 4x DS4 controllers (albeit, unsure of responsiveness!). I thought perhaps some of my cables were bogus, but have tested them all and they seem fine.

**Reply by darthcloud** [reply]
Its hard to tell but, you realise you can connect multiple cable to a single one?

Also I suspect you are doing this without soldering?

You might get away with it for a simple low wire build one using hard-coded FW.

But for universal that would be difficult.

Like connect all db25 gnd to the db25 connector shield (the metal piece) with short soldered wires. then connect one of the gc plug gnd to it (any is fine, one is ok, many is ok too) Connect one 5v to the power pin (or more doesn't matter).

Then connect all 4 data line.

**Comment by alastairbdoggett** [comment]
I am very much a novice wrt wiring so all tips you have I would be happy to hear :) Also I am trying to do this without soldering (using jump cables and breakout boards on the DB25, I have attached some photos of my materials and the current state of the project).

So from [this schematic](https://github.com/darthcloud/BlueRetroHW/blob/master/DIY/GameCube.pdf) I would need to solder 10x pins (14, 3, 5, 8 , ... , 24, 12) to the DB25 shield? Is there any reason why one would need to do this rather than what I am currently doing which is connecting the GND controller lines to these pins instead?

Universal Adapter
<img width="773" alt="Screen Shot 2021-10-23 at 11 13 34 PM" src="https://user-images.githubusercontent.com/17734670/138583155-901a4d30-21b3-4171-bc84-3370f16d7ddd.png">

GC Adapter
<img width="734" alt="Screen Shot 2021-10-23 at 11 14 43 PM" src="https://user-images.githubusercontent.com/17734670/138583181-1dfad459-ce16-474b-ba0f-ea90dc626adb.png">

**Reply by darthcloud** [reply]
It's just a suggestion on how to make it easier to manage multiple GND ;)

**Reply by alastairbdoggett** [reply]
Does the wiring look like it makes sense, in terms of the schematics? I'm trying to figure out if i've made a simple error somewhere in reading or interpreting the diagrams!

**Reply by darthcloud** [reply]
I can't see much from yours pictures, but my advice is to test each connection end to end with multimeter in continuity test mode (beeping mode).

Connect both your db25 together, and put a probe on a esp32  dev kit pin then make it beep by probing the other pin it should connect to on GC plug. Validate each pin this way.

Also all my connector diagram are facing the console, so if you look at controller plug directly, its mirrored.

**Reply by alastairbdoggett** [reply]
I managed to get it working in the end :)

Basically I had only grounded the 8x suggested pins (I39, IO21, IO22, IO25, IO32, IO33, I34 & I3) but neglected to ground the 3x remaining pins from [the schematic](https://github.com/darthcloud/BlueRetroHW/blob/master/DIY/GameCube.pdf) (namely 3, 5 & 8). I didn't think this was possible due to there only being 8x GND coming from the 4x GC cables but your suggestion of grouping multiple cables to one terminal told me otherwise!

So, just as you said, I needed *to connect all pins as described for console detection to work on universal build*, so these 3x GND cables that I missed were the issue :)

**Reply by darthcloud** [reply]
Congrats on getting it working, enjoy!

**Reply by alastairbdoggett** [reply]
thanks for the help! next up, psx/ps2 adapter!
